pkg/openstackbaremetalset/baremetalhost.go

```diff
@@ -70,8 +71,12 @@ func BaremetalHostProvision(
 	if userDataSecret == nil {
 		templateParameters := make(map[string]interface{})
 		templateParameters["AuthorizedKeys"] = strings.TrimSuffix(string(sshSecret.Data["authorized_keys"]), "\n")
-		templateParameters["Hostname"] = bmhStatus.Hostname
-		templateParameters["DomainName"] = instance.Spec.DomainName
+		templateParameters["HostName"] = bmhStatus.Hostname
+		//If Hostname is fqdn, use it
+		if !hostNameIsFQDN(bmhStatus.Hostname) && instance.Spec.DomainName != "" {
+			templateParameters["FQDN"] = strings.Join([]string{bmhStatus.Hostname, instance.Spec.DomainName}, ".")
+		}
+		templateParameters["FQDN"] = bmhStatus.Hostname
```

Comment:
Does there need to be an else here?  It seems like the template parametert that is set on L77 in the if-statement will always be overwritten on L79.

Reply:
If hostname is fqdn or not, we'll set it to fqdn like before, when `DomainName` is empty.

Reply:
sorry yeah you're right, my bad.